addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js

```diff
@@ -3,192 +3,167 @@
+        settingsPage.setAttribute("t-slot-scope", "settings");
 
-function compileBlock(el, params) {
-    const settingsContainer = createElement("SettingsBlock", {
-        title: toStringExpression(el.getAttribute("title") || ""),
-        tip: toStringExpression(el.getAttribute("help") || ""),
-    });
-    for (const child of el.children) {
-        append(settingsContainer, this.compileNode(child, params));
-    }
-    return settingsContainer;
-}
+        //props
+        params.modules = [];
 
-function compileSetting(el, params) {
-    const componentName = el.getAttribute("type") === "header" ? "SettingHeader" : "Setting";
-    const setting = createElement(componentName, {
-        title: toStringExpression(el.getAttribute("title") || ""),
-        help: toStringExpression(el.getAttribute("help") || ""),
-        companyDependent: el.getAttribute("company_dependent") === "1" || "false",
-        documentation: toStringExpression(el.getAttribute("documentation") || ""),
-        record: `this.props.record`,
-    });
-    let string = toStringExpression(el.getAttribute("string") || "");
-    let addLabel = true;
-    params.labels = [];
-    Array.from(el.children).forEach((child, index) => {
-        if (getTag(child, true) === "field" && index === 0) {
-            const fieldSlot = createElement("t", { "t-set-slot": "fieldSlot" });
-            const field = this.compileNode(child, params);
-            if (field) {
-                append(fieldSlot, field);
-                setting.setAttribute("fieldInfo", field.getAttribute("fieldInfo"));
-
-                addLabel = child.hasAttribute("nolabel")
-                    ? child.getAttribute("nolabel") !== "1"
-                    : true;
-                const fieldName = child.getAttribute("name");
-                string = child.hasAttribute("string")
-                    ? toStringExpression(child.getAttribute("string"))
-                    : string;
-                setting.setAttribute("fieldName", toStringExpression(fieldName));
-                setting.setAttribute(
-                    "fieldId",
-                    toStringExpression(child.getAttribute("field_id") || fieldName)
-                );
-            }
-            append(setting, fieldSlot);
-        } else {
-            append(setting, this.compileNode(child, params));
+        const res = super.compileForm(...arguments);
+        res.classList.remove("o_form_nosheet");
+
+        settingsPage.setAttribute("modules", JSON.stringify(params.modules));
+
+        for (const child of res.childNodes) {
+            append(settingsPage, this.compileNode(child, params));
         }
-    });
-    setting.setAttribute("string", string);
-    setting.setAttribute("addLabel", addLabel);
-    setting.setAttribute("labels", JSON.stringify(params.labels));
-    return setting;
-}
+        while (res.lastChild) {
+            res.removeChild(res.lastChild);
+        }
+        append(res, settingsPage);
```

Comment:
check this, looks like we compile children twice, could be nice to "simply" move them